receiver/awscontainerinsightreceiver/internal/k8swindows/extractors/extractorhelpers.go

```diff
@@ -1,9 +1,47 @@
 package extractors
 
 import (
+	"fmt"
+
 	stats "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
 )
 
+// convertCPUStats Convert kubelet CPU stats to Raw CPU stats
+func convertCPUStats(kubeletCPUStat *stats.CPUStats) *CPUStat {
+	var cpuStat CPUStat
```

Comment:
while we are null checking... should we null check the `kubeletCPUStat` pointer?

Reply:
yes, I am already checking null checks on kubeletCPUStat before func is called.

Reply:
the function should be self contained, we shouldn't rely on the calling context.

My two cents - avoid pointers when possible and just convert to/from pointers at the usage boundary.  It will also reduce our need for null checking.  I get that the kubeletCPUStat function has pointer member variables so we still need to check for nil before dereferencing.  What I am proposing is 
```
func convertCPUStats(kubeletCPUStat stats.CPUStats) CPUStat {
```
and
```
func convertMemoryStats(kubeletMemoryStat stats.MemoryStats) MemoryStat {
```

Performance wise, copying this struct is not expensive.
https://daryl-ng.medium.com/why-you-should-avoid-pointers-in-golang-998f7778c311

Reply:
Addressed. I have removed pointer references.
Thanks, this will definitely help avoid nil checks.